src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,271 @@
+use std::io::{Result, Write};
+
+use analysis::namespaces::MAIN;
+use env::Env;
+use file_saver::save_to_file;
+use library::*;
+use library::Type as LType;
+use writer::primitives;
+
+use stripper_interface as stripper;
+use stripper_interface::Type as SType;
+use stripper_interface::TypeStruct;
+use stripper_interface::{FILE, /*FILE_COMMENT, */MOD_COMMENT};
+
+pub trait ToStripperType {
+    fn convert(&self) -> stripper::TypeStruct;
+    fn doc(&self) -> Option<String>;
+    fn doc_deprecated(&self) -> Option<String>;
+}
+
+macro_rules! impl_to_stripper_type {
+    ($ty:ident, $enum_var:ident) => {
+        impl ToStripperType for $ty {
+            fn convert(&self) -> TypeStruct {
+                TypeStruct::new(SType::$enum_var, &self.name)
+            }
+
+            fn doc(&self) -> Option<String> {
+                self.doc.clone()
+            }
+
+            fn doc_deprecated(&self) -> Option<String> {
+                self.doc_deprecated.clone()
+            }
+        }
+    }
+}
+
+impl_to_stripper_type!(Alias, Struct);
+impl_to_stripper_type!(Constant, Const);
+impl_to_stripper_type!(Member, Variant);
+impl_to_stripper_type!(Enumeration, Enum);
+impl_to_stripper_type!(Bitfield, Type);
+impl_to_stripper_type!(Record, Type);
+impl_to_stripper_type!(Field, Variant);
+impl_to_stripper_type!(Union, Struct);
+impl_to_stripper_type!(Function, Fn);
+impl_to_stripper_type!(Interface, Trait);
+impl_to_stripper_type!(Class, Struct);
+impl_to_stripper_type!(Namespace, Mod);
+impl_to_stripper_type!(Parameter, Variant);
+
+pub fn generate(env: &Env) {
+    let path =  env.config.target_path.join("doc.cmts");
+    println!("Generating documentation {:?}", path);
+    save_to_file(&path, env.config.make_backup,
+        |w| generate_doc(w, &env.library));
+}
+
+fn generate_doc(mut w: &mut Write, lib: &Library) -> Result<()> {
+    let namespace = lib.namespace(MAIN);
+
+    /*if let Some(ref namespace_doc) = namespace.doc {
+        for line in namespace_doc.split("\n") {
+            output.push_str(&format!("{}{}\n", FILE_COMMENT, line));
+        }
+    }
+    try!(create_sub_docs(&mut output, &namespace.constants, None));*/
+    for ty in namespace.types.iter().filter_map(|t| t.as_ref()) {
+        try!(handle_type(&mut w, &ty, None, lib));
+    }
+    Ok(())
+}
+
+fn handle_type(w: &mut Write, ty: &LType, parent: Option<Box<TypeStruct>>, lib: &Library) -> Result<()> {
+    match *ty {
+        LType::Alias(ref a) => {
+            try!(writeln!(w, "{}src/auto/{}.rs", FILE, a.name.to_lowercase()));
```

Comment:
This is going to be fun because `GLArea` and GLContext` need module name overrides from config. We could leave this off for another PR though.